core/setup/kubernetes.go

```diff
@@ -0,0 +1,85 @@
+package setup
+
+import (
+//	"crypto/tls"
+//	"crypto/x509"
+    "fmt"
+//	"io/ioutil"
+//	"net"
+//	"net/http"
+//	"time"
+
+	"github.com/miekg/coredns/middleware"
+	"github.com/miekg/coredns/middleware/kubernetes"
+	k8sc "github.com/miekg/coredns/middleware/kubernetes/k8sclient"
+	"github.com/miekg/coredns/middleware/proxy"
+//	"github.com/miekg/coredns/middleware/singleflight"
+
+	"golang.org/x/net/context"
+)
+
+const defaultK8sEndpoint = "http://localhost:8080"
+
+// Kubernetes sets up the kubernetes middleware.
+func Kubernetes(c *Controller) (middleware.Middleware, error) {
+    fmt.Println("controller %v", c)
+    // TODO: Determine if subzone support required
+
+	kubernetes, err := kubernetesParse(c)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return func(next middleware.Handler) middleware.Handler {
+		kubernetes.Next = next
+		return kubernetes
+	}, nil
+}
+
+func kubernetesParse(c *Controller) (kubernetes.Kubernetes, error) {
+	k8s := kubernetes.Kubernetes{
+        Proxy:      proxy.New([]string{}),
+//		PathPrefix: "skydns",
+		Ctx:        context.Background(),
+//		Inflight:   &singleflight.Group{},
```

Comment:
Possibly not needed. Leaving in place until we do an optimization pass over the k8s connector.
I've added a comment to this effect.